### PR TITLE
feat: reuse Kafka proxy config logic

### LINKED ISSUE #43: Consider use of https://watermill.io for core routing and forwarding logic
https://watermill.io/ is an event-driven framework for Go. It is mainly used for reacting to events, however most of the logic behind that can potentially be used as base code for the message routing and forwarding of this event-gateway.

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR does all of the following:

- Unifies Kafka proxy config logic, specially around Kafka brokers address mapping.
- Adds descriptions to all configurations, so it improves documentation.
- It prints usage of env vars when an error is related to configuration.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -8,10 +8,10 @@ import (
 
 // App holds the config for the whole application.
 type App struct {
-	Debug        bool
-	AsyncAPIDoc  []byte      `split_words:"true"`
+	Debug        bool        `desc:"Enable or disable debug logs"`
+	AsyncAPIDoc  []byte      `split_words:"true" desc:"Path or URL to a valid AsyncAPI doc (v2.0.0 is only supported)"`
```
Comment (magicmatatjahu) [inline config/config.go:12]:
```suggestion
	AsyncAPIDoc  []byte      `split_words:"true" desc:"Path or URL to a valid AsyncAPI doc (v2.0.0 is only supported)"`
```

Only `2.0.0` or `2.1.0` also? :)

Comment (smoya) [reply config/config.go:12]: Unfortunately, 2.0.0 because go parser only parses up to that version.

Comment (magicmatatjahu) [reply config/config.go:12]: Ahhh... ok, I forgot about that.

**Review by magicmatatjahu — APPROVED** [review]
LGTM

**Pull request merged**